v2/eventsource/repository.go

```diff
@@ -283,6 +283,11 @@ func unmarshalRecords(serializer Serializer, records []Record) (events []Event,
 			err = errors.Wrap(err, "failed to unmarshal record")
 			return
 		}
+		// Some older events created with earlier releases did not have timestamp in
+		// record.Data so in those cases we pick up timestamp from event
+		if event.GetTimestamp() == int64(0) {
```

Comment:
Is this still a problem? Can we make a interceptor to fix this isssue?

Reply:
I have no idea, I just noticed this "fix" was added, I dont have any background information

Reply:
interceptor where?